functions/supplier/sftp-supplier/src/main/java/org/springframework/cloud/fn/supplier/sftp/SftpSupplierConfiguration.java

```diff
@@ -240,16 +251,24 @@ public RemoteFileDeletingAdvice remoteFileDeletingAdvice(SftpRemoteFileTemplate
 				MessageSource<?> sftpMessageSource,
 				MonoProcessor<?> subscriptionBarrier,
 				SftpSupplierProperties sftpSupplierProperties,
-				FileConsumerProperties fileConsumerProperties) {
+				FileConsumerProperties fileConsumerProperties,
+				@Nullable MessageHandler renameRemoteFileHandler) {
 
-			return FileUtils.enhanceFlowForReadingMode(IntegrationFlows
+			IntegrationFlowBuilder flowBuilder = FileUtils.enhanceFlowForReadingMode(IntegrationFlows
 							.from(IntegrationReactiveUtils.messageSourceToFlux(sftpMessageSource)
 									.delaySubscription(subscriptionBarrier)
 									.subscriberContext(
 											context -> (context.put(IntegrationReactiveUtils.DELAY_WHEN_EMPTY_KEY,
 													sftpSupplierProperties.getDelayWhenEmpty())))),
-					fileConsumerProperties)
-					.toReactivePublisher();
+					fileConsumerProperties);
+
+			if (renameRemoteFileHandler != null) {
+				flowBuilder.publishSubscribeChannel(pubsub ->
+						pubsub.subscribe(subFlow -> subFlow.handle(renameRemoteFileHandler).nullChannel())
```

Comment:
Why do we need here?
Doesn't look like we have a removal logic for `NonStreaming` variant...

Reply:
The removal logic is provided by SftpInboundAdapter for the non-streaming case, and as AFAIK there is no support for the renaming behavior we need in Spring Integration.

Reply:
Ah! Right, this one: `.deleteRemoteFiles(sftpSupplierProperties.isDeleteRemoteFiles())`. 
Sorry, good catch!

But yeah: rename is provided via gateway you do. So, all good. 👍